modules/core/src/test/java/org/apache/synapse/endpoints/RecipientListEndpointTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.synapse.endpoints;
+
+import org.apache.axiom.om.OMAbstractFactory;
+import org.apache.axiom.util.UIDGenerator;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.addressing.EndpointReference;
+import org.apache.axis2.clustering.Member;
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.axis2.context.OperationContext;
+import org.apache.axis2.context.ServiceContext;
+import org.apache.axis2.description.InOutAxisOperation;
+import org.apache.axis2.description.TransportInDescription;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.Entry;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.core.axis2.Axis2SynapseEnvironment;
+import org.apache.synapse.mediators.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.modules.junit4.PowerMockRunner;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ArrayList;
+
+
+
+/**
+ * Unit tests for class RecipientListEndpoint
+ */
+@RunWith(PowerMockRunner.class)
+public class RecipientListEndpointTest {
+
+    /**
+     * Test on init of RecipientListEndpoint
+     */
+    @Test
+    public void testRecipientListEndpointInit() {
+        try {
+            Axis2SynapseEnvironment synapseEnvironment = getMockedSynapseEnvironment();
+            RecipientListEndpoint recipientListEndpoint = new RecipientListEndpoint();
+
+            recipientListEndpoint.init(synapseEnvironment);
+        } catch (Exception ex) {
+            Assert.fail("recipientListEndpoint init must be successful without exception " + ex);
```

Comment:
Can't we avoid catching the exception? The test will look cleaner that way

Reply:
You mean catching the specific exception?